lib/APMSpan.js

```diff
@@ -365,10 +372,31 @@ class APMSpan {
         // only parent span should be recorded -> one Trace with all nodes
         const spanToReport = this[SPAN_CONTEXT].getTrace().isFinished(this);
         if (spanToReport) {
-            this[TRACER].getRecorder().record(spanToReport);
+            const level = this[SPAN_CONTEXT].getLevel() === REPORTING_LEVEL_ALL ? this[SPAN_CONTEXT].getLevel() :
+                spanToReport.context().getLevel();
+            const isSampled = this.contextSampling(this[TRACER].getSampler(), this[SPAN_CONTEXT].getTrace(), level);
+
+            if (isSampled) {
+                this[TRACER].getRecorder().record(spanToReport);
+            }
         }
     }
 
+    contextSampling(sampler, trace, reportingLevel) {
+        if (reportingLevel) {
+            switch (reportingLevel) {
+            case REPORTING_LEVEL_NONE:
+            case REPORTING_LEVEL_IGNORE:
+                return false;
+            case REPORTING_LEVEL_ALL:
+            default:
+                return true;
+            }
```

Comment:
you have a small typo there:
` return reportingTypes.hasOwnProperty(reportingLevel) ? reportingTypes[reportingLevel] : true;`

Well in some cases it's nicer to use `object literals`. I will stick with switch statement here.